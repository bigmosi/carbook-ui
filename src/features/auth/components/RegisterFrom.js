import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate, Link } from 'react-router-dom';
import { registerAsync, selectErrors } from '../authSlice';

const LoginFrom = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [username, setUsername] = useState();
  const [password, setPassword] = useState();
  const [passwordConfirmation, setPasswordConfirmation] = useState();
  const errors = useSelector(selectErrors);

  const onSubmit = () => {
    const user = {
      username,
      password,
      password_confirmation: passwordConfirmation,
    };

    dispatch(registerAsync(user)).unwrap()
      .then(() => navigate('/login'));
  };

  return (
    <div>
      <form className="form">
        <div>
          <label className="label" htmlFor="username">
            Username
            <input className="input" id="username" onChange={(e) => setUsername(e.target.value)} />
          </label>
          <div className="input-error">{errors?.username?.length ? errors.username[0] : errors.username }</div>
        </div>
        <div>
          <label className="label" htmlFor="password">
            Password
            <input className="input" id="password" type="password" onChange={(e) => setPassword(e.target.value)} />
          </label>
          <div className="input-error">{errors?.password?.length ? errors.password[0] : errors.password }</div>
        </div>
        <div>
          <label className="label" htmlFor="password_confirmation">
            Password confirmation
            <input className="input" id="password_confirmation" type="password" onChange={(e) => setPasswordConfirmation(e.target.value)} />
          </label>
          <div className="input-error">{errors?.password_confirmation?.length ? errors.password_confirmation[0] : errors.password_confirmation }</div>
        </div>
        <div>
          <button className="button" type="button" onClick={onSubmit}>Register</button>
        </div>
      </form>
      <small className="auth-footer">
        <span>Already have an account? </span>
        <Link className="text-indigo-600" to="/login">Login</Link>
      </small>
    </div>
  );
};

export default LoginFrom;
