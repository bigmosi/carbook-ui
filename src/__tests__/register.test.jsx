import React from 'react';
import '@testing-library/jest-dom';
import renderWithProviders from '../utils/test-utils';
import RegisterFrom from '../features/auth/components/RegisterFrom';

describe('Register component', () => {
  const { asFragment } = renderWithProviders(<RegisterFrom />);
  const firstRender = asFragment();

  it('should render correctly', () => {
    expect(firstRender).toMatchSnapshot();
  });
});
