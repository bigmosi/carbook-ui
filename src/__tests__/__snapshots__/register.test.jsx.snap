// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <form
      class="form"
    >
      <div>
        <label
          class="label"
          for="username"
        >
          Username
          <input
            class="input"
            id="username"
          />
        </label>
      </div>
      <div>
        <label
          class="label"
          for="password"
        >
          Password
          <input
            class="input"
            id="password"
            type="password"
          />
        </label>
      </div>
      <div>
        <button
          class="button"
          type="button"
        >
          Log in
        </button>
      </div>
    </form>
    <small
      class="auth-footer"
    >
      <span>
        Dont have an account? 
      </span>
      <a
        href="/register"
      >
        Regiter
      </a>
    </small>
  </div>
</DocumentFragment>
`;

exports[`Register component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <form
      class="form"
    >
      <div>
        <label
          class="label"
          for="username"
        >
          Username
          <input
            class="input"
            id="username"
          />
        </label>
        <div
          class="input-error"
        />
      </div>
      <div>
        <label
          class="label"
          for="password"
        >
          Password
          <input
            class="input"
            id="password"
            type="password"
          />
        </label>
        <div
          class="input-error"
        />
      </div>
      <div>
        <label
          class="label"
          for="password_confirmation"
        >
          Password confirmation
          <input
            class="input"
            id="password_confirmation"
            type="password"
          />
        </label>
        <div
          class="input-error"
        />
      </div>
      <div>
        <button
          class="button"
          type="button"
        >
          Register
        </button>
      </div>
    </form>
    <small
      class="auth-footer"
    >
      <span>
        Already have an account? 
      </span>
      <a
        href="/login"
      >
        Login
      </a>
    </small>
  </div>
</DocumentFragment>
`;
